lib/llm/src/preprocessor.rs

```diff
@@ -77,6 +78,15 @@ pub struct JailState {
     finished: bool,                            // Add this flag to track if stream is finished
 }
 
+pub fn enable_tool_call(parser_str: Option<&str>, request: &NvCreateChatCompletionRequest) -> bool {
+    // Enable tool call if the below two conditions are satisfied
+    // 1. parser_str is not None
+    // 2. tool_choice is not None
+    parser_str.is_some()
+        && request.inner.tool_choice.is_some()
+        && (request.inner.tool_choice.clone() != Some(ChatCompletionToolChoiceOption::None))
```

Comment:
`&& matches!(request.inner.tool_choice, Some(ChatThingy::None))` instead of these two lines.

`matches!` is a macro that expands to a single line `match xy` statement.

Even if you don't use that, you should't need the `clone` here.